pkg/testutils/integration_test.go

```diff
@@ -0,0 +1,106 @@
+package testutils_test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/Azure/adx-mon/pkg/testutils"
+	"github.com/Azure/adx-mon/pkg/testutils/collector"
+	"github.com/Azure/adx-mon/pkg/testutils/ingestor"
+	"github.com/Azure/adx-mon/pkg/testutils/kustainer"
+	"github.com/stretchr/testify/require"
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/modules/k3s"
+)
+
+func TestIntegration(t *testing.T) {
+	// An extra generous timeout for the test. The test should run in
+	// about 5 minutes, but when running with the race detector, it
+	// can take longer.
+	ctx, cancel := context.WithTimeout(context.Background(), time.Hour)
+	defer cancel()
+
+	k3sContainer, err := k3s.Run(ctx, "rancher/k3s:v1.31.2-k3s1")
+	testcontainers.CleanupContainer(t, k3sContainer)
+	require.NoError(t, err)
+
+	kustoContainer, err := kustainer.Run(ctx, "mcr.microsoft.com/azuredataexplorer/kustainer-linux:latest", kustainer.WithCluster(ctx, k3sContainer))
+	testcontainers.CleanupContainer(t, kustoContainer)
+	require.NoError(t, err)
+
+	restConfig, _, err := testutils.GetKubeConfig(ctx, k3sContainer)
+	require.NoError(t, err)
+	require.NoError(t, kustoContainer.PortForward(ctx, restConfig))
+
+	kubeconfig, err := testutils.WriteKubeConfig(ctx, k3sContainer, t.TempDir())
+	require.NoError(t, err)
+	t.Logf("Kubeconfig: %s", kubeconfig)
+	t.Logf("Kustainer: %s", kustoContainer.ConnectionUrl())
+
+	t.Run("Create databases", func(t *testing.T) {
+		for _, dbName := range []string{"Metrics", "Logs"} {
+			require.NoError(t, kustoContainer.CreateDatabase(ctx, dbName))
+		}
+	})
+
+	ingestorContainer, err := ingestor.Run(ctx, ingestor.WithCluster(ctx, k3sContainer))
```

Comment:
For a follow on PR, I think it would be nice to have a way to configure the Kusto batch policies so we can set them to something like 30s for logs (instead of the 5m we hard code them to) which hopefully speeds this test up a lot.

Reply:
Added this functionality, which was a bit more complicated than I would have thought due to the time formatting required by Kusto.